skimage/morphology/_skeletonize_cy.pyx

```diff
@@ -74,39 +74,42 @@ def _fast_skeletonize(image):
 
     # the algorithm reiterates the thinning till
     # no further thinning occurred (variable pixel_removed set)
-
-    while pixel_removed:
-        pixel_removed = False
-
-        # there are two phases, in the first phase, pixels labeled (see below)
-        # 1 and 3 are removed, in the second 2 and 3
-
-        for first_pass in (True, False):
-            for row in range(1, nrows-1):
-                for col in range(1, ncols-1):
-                    # all set pixels ...
-                    if skeleton[row, col]:
-                        # are correlated with a kernel (coefficients spread around here ...)
-                        # to apply a unique number to every possible neighborhood ...
-
-                        # which is used with the lut to find the "connectivity type"
-
-                        neighbors = lut[  1*skeleton[row - 1, col - 1] +   2*skeleton[row - 1, col] +\
-                                          4*skeleton[row - 1, col + 1] +   8*skeleton[row, col + 1] +\
-                                         16*skeleton[row + 1, col + 1] +  32*skeleton[row + 1, col] +\
-                                         64*skeleton[row + 1, col - 1] + 128*skeleton[row, col - 1]]
-
-                        # if the condition is met, the pixel is removed (unset)
-                        if ((neighbors == 1 and first_pass) or
-                                (neighbors == 2 and not first_pass) or
-                                (neighbors == 3)):
-                            cleaned_skeleton[row, col] = 0
-                            pixel_removed = True
-
-            # once a step has been processed, the original skeleton
-            # is overwritten with the cleaned version
-            _skeleton = _cleaned_skeleton.copy()
-            skeleton = _skeleton
+    with nogil:
+        while pixel_removed:
+            pixel_removed = False
+
+            # there are two phases, in the first phase, pixels labeled (see below)
+            # 1 and 3 are removed, in the second 2 and 3
+
+            # nogil can't iterate through `(True, False)` because it is a Python
+            # tuple. Use the fact that 0 is Falsy, and 1 is truthy in C
+            # for the iteration instead.
+            # for first_pass in (True, False):
+            for first_pass in range(1, -1, -1):
```

Comment:
OOOOOH. I get this now. I'd read it as `for first_pass in (1, -1, -1)` (no `range`).

Yeah nah. =P I find this terribly unreadable. How about:

```suggestion
            for pass_num in range(2):
                first_pass = (pass_num == 0)
```

Reply:
I’m not sure. The goal was to not change the order in which true and false were iterated in. Who knows if it makes a difference. I’m not the original author. 

If you want. Ill accept your suggestion.

Reply:
Lets see what the cis say

Reply:
@hmaarrfk the booleans are traversed in the same order this way, it just takes two lines instead of one. =)

... The CIs appear very unhappy indeed... LOL

Reply:
... aaaaand it's a new release of pytest that breaks everything. :man_facepalming::man_facepalming::man_facepalming::man_facepalming::man_facepalming:

Reply:
I think I had a pr that tried to address some pytest deprecations. Maybe you can relaunch that one.